File: frontend/lib/features/intent_analysis/views/intent_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../providers/intent_provider.dart';
import '../../../shared/widgets/loading_state_view.dart';
import '../../../shared/widgets/error_state_view.dart';

class IntentScreen extends ConsumerStatefulWidget {
  const IntentScreen({super.key});

  @override
  ConsumerState<IntentScreen> createState() => _IntentScreenState();
}

class _IntentScreenState extends ConsumerState<IntentScreen> {
  final _controller = TextEditingController();

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final intentState = ref.watch(intentProvider);

    return Scaffold(
      appBar: AppBar(title: const Text('Intent Analysis')),
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          children: [
            TextField(
              controller: _controller,
              decoration: const InputDecoration(
                labelText: 'Paste message or transcript here...',
                border: OutlineInputBorder(),
              ),
              maxLines: 5,
            ),
            const SizedBox(height: 16),
            ElevatedButton.icon(
              onPressed: () {
                FocusScope.of(context).unfocus();
                ref.read(intentProvider.notifier).analyzeIntent(_controller.text);
              },
              icon: const Icon(Icons.analytics),
              label: const Text('Analyze Intent'),
            ),
            const SizedBox(height: 24),
            Expanded(
              child: intentState.when(
                data: (response) {
                  if (response == null) return const SizedBox.shrink();
                  final isScam = response.isScam;
                  return Card(
                    color: isScam ? Colors.red.shade50 : Colors.green.shade50,
                    child: Padding(
                      padding: const EdgeInsets.all(16.0),
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.stretch,
                        children: [
                          const SizedBox(height: 16),
                          Text(
                            isScam ? '⚠️ SCAM DETECTED' : '✅ SAFE',
                            style: Theme.of(context).textTheme.headlineSmall?.copyWith(
                              color: isScam ? Colors.red : Colors.green,
                              fontWeight: FontWeight.bold,
                            ),
                            textAlign: TextAlign.center,
                          ),
                          const SizedBox(height: 16),
                          Text(
                            'Scam Score: ${response.scamScore}/100',
                            style: Theme.of(context).textTheme.titleMedium,
                            textAlign: TextAlign.center,
                          ),
                          const Divider(),
                          Text(
                            response.reason,
                            style: Theme.of(context).textTheme.bodyMedium,
                            textAlign: TextAlign.center,
                          ),
                        ],
                      ),
                    ),
                  );
                },
                loading: () => const LoadingStateView(message: 'Analyzing intent...'),
                error: (error, _) => ErrorStateView(
                  errorMessage: error.toString(),
                  onRetry: () => ref.read(intentProvider.notifier).analyzeIntent(_controller.text),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: frontend/app/src/main/java/com/aegis/scamfirewall/features/intent/IntentScreen.kt
package com.aegis.scamfirewall.features.intent

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Analytics
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.aegis.scamfirewall.core.network.ApiService
import com.aegis.scamfirewall.core.theme.AlertGreenBg
import com.aegis.scamfirewall.core.theme.AlertRedBg
import com.aegis.scamfirewall.models.IntentResponse
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun IntentScreen(
    apiService: ApiService,
    onBack: () -> Unit
) {
    var transcript by remember { mutableStateOf("") }
    var isLoading by remember { mutableStateOf(false) }
    var result by remember { mutableStateOf<IntentResponse?>(null) }
    var error by remember { mutableStateOf<String?>(null) }
    val scope = rememberCoroutineScope()

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Intent Analysis") },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = "Back"
                        )
                    }
                }
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(16.dp)
                .verticalScroll(rememberScrollState()),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            OutlinedTextField(
                value = transcript,
                onValueChange = { transcript = it },
                label = { Text("Paste message or transcript here...") },
                modifier = Modifier
                    .fillMaxWidth()
                    .height(150.dp),
                maxLines = 6
            )
            Spacer(modifier = Modifier.height(16.dp))
            Button(
                onClick = {
                    if (transcript.isNotBlank()) {
                        scope.launch {
                            isLoading = true
                            error = null
                            result = null
                            try {
                                result = apiService.analyzeIntent(transcript)
                            } catch (e: Exception) {
                                error = e.localizedMessage ?: "Failed to analyze transcript"
                            } finally {
                                isLoading = false
                            }
                        }
                    }
                },
                enabled = transcript.isNotBlank() && !isLoading,
                modifier = Modifier.fillMaxWidth()
            ) {
                if (isLoading) {
                    CircularProgressIndicator(
                        modifier = Modifier.size(24.dp),
                        color = MaterialTheme.colorScheme.onPrimary,
                        strokeWidth = 2.dp
                    )
                } else {
                    Icon(imageVector = Icons.Default.Analytics, contentDescription = null)
                    Spacer(modifier = Modifier.width(8.dp))
                    Text("Analyze Intent")
                }
            }
            
            Spacer(modifier = Modifier.height(24.dp))

            if (error != null) {
                Card(
                    colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.errorContainer),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Column(modifier = Modifier.padding(16.dp)) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(imageVector = Icons.Default.Warning, contentDescription = "Error")
                            Spacer(modifier = Modifier.width(8.dp))
                            Text(text = "Error", fontWeight = FontWeight.Bold)
                        }
                        Spacer(modifier = Modifier.height(8.dp))
                        Text(text = error!!)
                    }
                }
            }

            if (result != null) {
                val response = result!!
                val isScam = response.isScam
                
                Card(
                    colors = CardDefaults.cardColors(
                        containerColor = if (isScam) AlertRedBg else AlertGreenBg
                    ),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Column(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(24.dp),
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        Text(
                            text = if (isScam) "⚠️ SCAM DETECTED" else "✅ SAFE",
                            style = MaterialTheme.typography.headlineMedium,
                            fontWeight = FontWeight.Bold,
                            color = if (isScam) Color.Red else Color.Green,
                            textAlign = TextAlign.Center
                        )
                        Spacer(modifier = Modifier.height(16.dp))
                        Text(
                            text = "Scam Score: ${response.scamScore}/100",
                            style = MaterialTheme.typography.titleMedium,
                            fontWeight = FontWeight.SemiBold,
                            textAlign = TextAlign.Center
                        )
                        Spacer(modifier = Modifier.height(8.dp))
                        HorizontalDivider(
                            modifier = Modifier.padding(vertical = 8.dp),
                            color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.2f)
                        )
                        Text(
                            text = response.reason,
                            style = MaterialTheme.typography.bodyMedium,
                            textAlign = TextAlign.Center
                        )
                    }
                }
            }
        }
    }
}
